### PR TITLE
feat: eventing-in testkit for protocol-first

### PR DESCRIPTION
**Pull request by efgpinto**
Refs https://github.com/lightbend/kalix/issues/7883

- [x] Java basic API
- [x] Scala basic API

### CODE REVIEW

File: samples/java-protobuf-eventsourced-counter/src/it/java/com/example/CounterTopicSubscriptionIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    var msg = CounterTopicApi.Increased.newBuilder().setValue(15).build();
+    var md = Metadata.EMPTY
+        .add("Content-Type", "application/protobuf;proto="+ CounterTopicApi.Increased.getDescriptor().getFullName())
+        .add("ce-specversion", "1.0")
+        .add("ce-id", "msg1")
+        .add("ce-type", "Increase")
+        .add("ce-source", CounterTopicApi.Increased.getDescriptor().getFullName())
+        .add("ce-subject", "counter-2");
+
+    topic.publish(msg.toByteString(), md);
```
Comment (johanandren) [inline samples/java-protobuf-eventsourced-counter/src/it/java/com/example/CounterTopicSubscriptionIntegrationTest.java:73]: We should accept generated message and byte-string + add type to metadata ourselves. Possibly also require subject id as a string parameter and not require the user to deal with metadata at all unless they want to add additional meta.

File: sdk/java-sdk-protobuf-testkit/src/main/scala/kalix/javasdk/testkit/impl/EventingTestKitImpl.scala [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+    override def eventStreamOut(in: Source[EventStreamOutCommand, NotUsed]): Source[EventStreamOutResult, NotUsed] = ???
+  }
+}
+
+final class TopicImpl(private val destinationProbe: TestProbe, private val sourceProbe: TestProbe) extends Topic {
+
+  private lazy val brokerProbe = sourceProbe.expectMsgType[RunningSourceProbe]
+  private val log = LoggerFactory.getLogger(classOf[TopicImpl])
+
+  @ApiMayChange
```
Comment (johanandren) [inline sdk/java-sdk-protobuf-testkit/src/main/scala/kalix/javasdk/testkit/impl/EventingTestKitImpl.scala:229]: No need for @ApiMayChange in internals `impl`, we can change anything here (but top types should be marked internal api)

File: samples/java-protobuf-eventsourced-counter/src/it/java/com/example/CounterTopicSubscriptionIntegrationTest.java (hunk 1 above)
Comment (efgpinto) [reply samples/java-protobuf-eventsourced-counter/src/it/java/com/example/CounterTopicSubscriptionIntegrationTest.java]: thanks, added now in https://github.com/lightbend/kalix-jvm-sdk/pull/1621/commits/58fdab1a978de431a74994753065a7c3c48a82ea

**Review by efgpinto — COMMENTED** [review]
Ready for new review. Includes both scala and java api for protocol-first.

File: samples/java-protobuf-eventsourced-counter/src/it/java/com/example/CounterTopicIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  private final EventingTestKit.Topic eventsTopic;
+
+
+  public CounterTopicIntegrationTest() {
+    client = testKit.getGrpcClient(CounterService.class);
+    commandsTopic = testKit.getTopic("counter-commands");
+    eventsTopic = testKit.getTopic("counter-events");
+  }
+
+  private int getCounterValue(String counterId) throws Exception {
```
Comment (aludwiko) [inline samples/java-protobuf-eventsourced-counter/src/it/java/com/example/CounterTopicIntegrationTest.java:44]: minor, I would move private methods at the bottom

File: samples/java-protobuf-eventsourced-counter/src/it/java/com/example/CounterTopicIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    var decreaseCmd = CounterApi.DecreaseValue.newBuilder().setCounterId(counterId).setValue(1).build();
+    commandsTopic.publish(increaseCmd, counterId);
+    commandsTopic.publish(decreaseCmd, counterId);
+
+    var increasedEvent = eventsTopic.expectOneTyped(CounterTopicApi.Increased.class);
+    var decreasedEvent = eventsTopic.expectOneTyped(CounterTopicApi.Decreased.class);
+    assertEquals(increaseCmd.getValue(), increasedEvent.getPayload().getValue());
+    assertEquals(decreaseCmd.getValue(), decreasedEvent.getPayload().getValue());
+
+    assertEquals(3, getCounterValue(counterId));
```
Comment (aludwiko) [inline samples/java-protobuf-eventsourced-counter/src/it/java/com/example/CounterTopicIntegrationTest.java:66]: this assertion is not present in the Scala sample, maybe we could remove from here?

**Review by aludwiko — APPROVED** [review]
some minors only

**Pull request merged**